### DISCUSSION TITLE
Messages with same ID (Duplication), take the new one

### DISCUSSION THREAD

**Discussion by fabianhaaa** [nats-io/nats.java#633] [Q&A] [answered]
Hey, following example:
I have a list of 3 Messages, the first one is the oldest and the last is the newest.
All 3 Messages have the same ID, so when i push them nats detecting it as duplication which is right. If iam now subscribing to it, i only get the oldest Message. Is there any solution that i can get the newest one ? Or do i must go trough the list before iam pushing it and delete the old ones ?

**Comment by scottf** [comment, marked as answer]
Messages published with the duplicate message id are not added to the stream, which is why you always get the "oldest". Maybe KeyValueStore will solve your problem, you can continually publish to the same key and the value of the key is always the latest, and you can configure how much history per key is retained.
